### PR TITLE
do not trigger mass versioning when stage is created

### PR DESCRIPTION
**Pull request by grosser**
@jonmoter 

 - fixing original bug
 - cleaning up duplicate versions
 - making future bugs easy to spot

JIRA: https://zendesk.atlassian.net/browse/SAMSON-296

### CODE REVIEW

File: db/migrate/20161026004331_remove_duplicate_versions.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+class RemoveDuplicateVersions < ActiveRecord::Migration[5.0]
+  def up
+    # delete all versions that are not unique
+    PaperTrail::Version.connection.execute <<-SQL
```
Comment (grosser) [inline db/migrate/20161026004331_remove_duplicate_versions.rb]: this surprisingly works on mysql/sqlite/postgres

File: db/migrate/20161026004331_remove_duplicate_versions.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      delete from versions where id not in (
+        SELECT * FROM (select min(id) from versions group by item_id, item_type, object) AS temp_tab
+      )
```
Comment (jonmoter) [inline db/migrate/20161026004331_remove_duplicate_versions.rb]: Have you tried running a variation of this query to find out how many rows would be deleted?

**Comment by jonmoter** [comment]
@pswadi-zendesk, this is a fix for a bug where PaperTrail was creating thousands of records due to an interaction between the `Stage` and `Commands` table.

Since this PR includes a migration that will delete a bunch of data from the `Versions` table in our database, do we need to get some sort of approval before running it?

**Comment by pswadi-zendesk** [comment]
Can we find out what rows will be deleted before hand? Also, lets take a backup of the DB so we have it for audit. I will talk with security to confirm.

**Comment by grosser** [comment]
```
PaperTrail::Version.count_by_sql <<-SQL
  select count(id) from versions where id not in (
    SELECT * FROM (select min(id) from versions group by item_id, item_type, object) AS temp_tab
  )
SQL

=> 451679
```
of ~500k
I can also dump them all into a file and if we see the need for it we can go through that ?
... was assuming we have regular db backup anyway ...

**Pull request merged**